kratos/utilities/mortar_utilities.h

```diff
@@ -592,22 +600,328 @@ class MortarUtilities
         /* DEFINITIONS */        
         bounded_matrix<double, TNumNodes, TDim> AbsMatrix;
         
-        for (unsigned int i_node = 0; i_node < TNumNodes; i_node++)
+        for (unsigned int i_node = 0; i_node < TNumNodes; ++i_node)
         {
-            for (unsigned int i_dof = 0; i_dof < TDim; i_dof++)
+            for (unsigned int i_dof = 0; i_dof < TDim; ++i_dof)
             {
                 AbsMatrix(i_node, i_dof) = std::abs(InputMatrix(i_node, i_dof));
             }
         }
         
         return AbsMatrix;
     }
+    /**
+     * This method resets the value
+     * @param ThisGeometry: The geometrty to update
+     * @param ThisVariable: The variable to set
+     */
+    template< class TVarType, HistoricalValues THist>
+    static inline void ResetValue(
+        GeometryType& ThisGeometry,
+        TVarType& ThisVariable
+        );
+    
+    /**
+     * This method adds the value
+     * @param ThisGeometry: The geometrty to update
+     * @param ThisVariable: The variable to set
+     */
+    template< class TVarType, HistoricalValues THist>
+    static inline void MatrixValue(
+        GeometryType& ThisGeometry,
+        TVarType& ThisVariable,
+        Matrix& ThisValue
+        );
     
+    /**
+     * This method adds the value
+     * @param ThisGeometry: The geometrty to update
+     * @param ThisVariable: The variable to set
+     */
+    template< class TVarType, HistoricalValues THist>
+    static inline void AddValue(
+        GeometryType& ThisGeometry,
+        TVarType& ThisVariable,
+        const Matrix& ThisValue
+        );
+
 private:
 };// class MortarUtilities
 
 ///@name Explicit Specializations
 ///@{
 
+template<> 
+inline void MortarUtilities::ResetValue<Variable<double>, Historical>(
+        GeometryType& ThisGeometry,
+        Variable<double>& ThisVariable
+        )
+{
+    for (unsigned int i_node = 0; i_node < ThisGeometry.size(); ++i_node)
+    {
+        ThisGeometry[i_node].FastGetSolutionStepValue(ThisVariable) = 0.0;
+    }
+}
+
+template<> 
+inline void MortarUtilities::ResetValue<component_type, Historical>(
+        GeometryType& ThisGeometry,
+        component_type& ThisVariable
+        )
+{
+    for (unsigned int i_node = 0; i_node < ThisGeometry.size(); ++i_node)
+    {
+        ThisGeometry[i_node].FastGetSolutionStepValue(ThisVariable) = 0.0;
+    }
+}
+
+template<> 
```

Comment:
make a comment that this is not threadsafe. It is really easy to use this wrong in an openmp loop

Reply:
Ok